File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        Map<String, Integer> wantMap = new HashMap<>();

        // Map에 모두 넣는다
        for (int i = 0; i < want.length; i++) {
            wantMap.put(want[i], number[i]);
        }

        for (int i = 0; i < discount.length; i++) {
            if (i == discount.length - 9)
                break;
            Map<String, Integer> discoutMap = new HashMap<>();
            boolean isCorrect = true;
            for (int j = i; j < i + 10; j++) {
                discoutMap.put(discount[j], discoutMap.getOrDefault(discount[j], 0) + 1);
            }

            for (int j = 0; j < wantMap.size(); j++) {
                if (wantMap.get(want[j]) != discoutMap.get(want[j])) {
                    isCorrect = false;
                    break;
                }
            }

            if (isCorrect)
                answer++;
        }
        return answer;
    }
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    
    wantDict = dict()
    # dictionary 사용
    for i in range(len(want)) :
        wantDict[want[i]] = number[i]
    
    for i in range(len(discount)) :
        check = True
        if i == len(discount) - 9 :
            break
        
        discountDict = dict()
        for j in range(i, i + 10) :
            discountDict[discount[j]] = discountDict.get(discount[j], 0) + 1
            
        for j in range(len(want)) :
            if wantDict.get(want[j], -1) !=  discountDict.get(want[j] , 0) :
                check = False
                break
        
        if check :
            answer += 1
            
    return answer